chef/cookbooks/bind9/recipes/default.rb

```diff
@@ -247,8 +245,8 @@ def make_zone(zone)
   next unless network.key?("allocated_by_name")
   net_name=network[:id].gsub(/_network$/, "").gsub("_","-")
   network[:allocated_by_name].each_key do |host|
-    if search(:node, "fqdn:#{host}").size > 0 or not host.match(/.#{node[:dns][:domain]}$/)
-      #this is node in crowbar terms or it not belong to our domain, so lets skip it
+    if !host.match(/.#{node[:dns][:domain]}$/) || fqdns.include?(host)
+      # this is node in crowbar terms or it not belong to our domain, so lets skip it
```

Comment:
I don't quite understand this comment TBH, maybe I'm lacking context. Does it want to say that the node either belongs to crowbar or is out of our domain?